Source/Models/JSONDeserializer.swift

```diff
@@ -0,0 +1,36 @@
+import Argo
+import Result
+
+struct JSONDeserializer: Deserializer {
+  func deserialize(data: NSData?) -> Result<AnyObject, SwishError> {
+    let json = self.parseJSON(data)
+
+    switch json {
```

Comment:
you know, this could probably use `analysis` instead of a switch:

```swift
return json.analysis(
  ifSuccess(.Success),
  ifFailure { .Failure(.InvalidJSONResponse($0)) } // could obviously write a helper function for this
)
```

might need More Type Info, but just a thought.